common/schema/go2proto.to.go

```diff
@@ -86,14 +174,22 @@ func AWSIAMAuthDatabaseConnectorFromProto(v *destpb.AWSIAMAuthDatabaseConnector)
 	}
 
 	out = &AWSIAMAuthDatabaseConnector{}
-	if fieldPos, err := PositionFromProto(v.Pos); err != nil {
+	out.Pos, err = orZeroR(toResult(PositionFromProto(v.Pos))).result()
+	if err != nil {
 		return nil, fmt.Errorf("Pos: %w", err)
-	} else {
-		out.Pos = fromPtr(fieldPos)
 	}
-	out.Username = string(v.Username)
-	out.Endpoint = string(v.Endpoint)
-	out.Database = string(v.Database)
+	out.Username, err = orZeroR(toResult(ptr(string(v.Username)), nil)).result()
```

Comment:
Mmmm.... this generated code seems much much grosser :\

Reply:
Yeah, That is the tradeoff with generalising the pointers / optionals / slices from the underlying type conversion machinery 😞 

We could optimise this in the future by constructing AST struct of the transformations and optimising that before producing the code